clients/drcachesim/tracer/instr_counter.cpp

```diff
@@ -550,9 +550,9 @@ compute_irregular_trace_windows(std::vector<instr_interval_t> &instr_intervals)
     // must have a duration long enough to cover the end of the program.
     irregular_window_ptr =
         (irregular_window_t *)dr_global_alloc(sizeof(irregular_window_t));
-    // DELAY_FOREVER_THRESHOLD might be too small for long traces, but it doesn't matter
-    // because we trace_for_instrs = 0, so no window is created anyway.
-    irregular_window_ptr->no_trace_for_instrs = DELAY_FOREVER_THRESHOLD;
+    // This results in has_instr_count_threshold_to_enable_tracing() returning 0, which
+    // in turns makes hit_instr_count_threshold() return without creating a new window.
+    irregular_window_ptr->no_trace_for_instrs = 0;
```

Comment:
Apologies if this was already considered: can we potentially simplify this by checking if `irregular_window_idx >= num_irregular_windows` (instead of adding the extra last entry)?

Reply:
Yes, we can check for that in `get_current_no_trace_for_instrs_value()` and `get_current_trace_for_instrs_value()` and return 0 there.
I don't have a strong preference. In one case we're doing an extra check for every call to `get_current_[no_]trace_for_instrs_value()`, but we'd avoid having this last additional window.

Fundamentally it doesn't change the rest of the logic though.